[CI] JdbcSecurityIT class failing

**Build Scans:**
- [elasticsearch-periodic #12339 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/ooxllj5k7j4m6)
- [elasticsearch-periodic-platform-support #12306 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/qybagmqqapgju)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:sql:qa:server:security:without-ssl:javaRestTest" --tests "org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT.testQuerySingleFieldExceptionedWrongRequested" -Dtests.seed=461AE6B5F37E56DC -Dtests.jvm.argline="-Des.concurrent_search=true" -Dtests.locale=cv-RU -Dtests.timezone=America/Dominica -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Previous test had an audit-related failure. All subsequent audit related assertions are bogus because we can't guarantee that we fully cleaned up after the last test.
```

**Issue Reasons:**
- [main] 2 failures in class org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT (1.1% fail rate in 176 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/SqlSecurityTestCase.java
+++ b/x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/SqlSecurityTestCase.java
@@ -37,6 +37,7 @@ import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
@@ -157,8 +158,18 @@ public abstract class SqlSecurityTestCase extends ESRestTestCase {
     }
 
     @Before
-    public void setInitialAuditLogOffset() throws IOException {
-        auditLogLinesBeforeTestStart = countAuditLogLines();
+    public void setInitialAuditLogOffset() throws Exception {
+        // Audit events are flushed asynchronously, so lines from the previous test may still be
+        // arriving when this test starts. Wait until the audit log stops growing before recording
+        // the offset, otherwise late-arriving lines from a prior test would land inside this test's
+        // window and be treated as unmatched logs.
+        assertBusy(() -> {
+            int first = countAuditLogLines();
+            Thread.sleep(100);
+            int second = countAuditLogLines();
+            assertEquals("audit log still being written", first, second);
+            auditLogLinesBeforeTestStart = second;
+        });
     }
 
     private int countAuditLogLines() throws IOException {
@@ -513,6 +524,7 @@ public abstract class SqlSecurityTestCase extends ESRestTestCase {
                 auditFailure
             );
             try {
+                // Audit events flush asynchronously; allow generous time for the expected events to appear.
                 assertBusy(() -> {
                     List<String> allLines;
                     try (InputStream auditLog = getCluster().getNodeLog(0, LogType.AUDIT)) {
@@ -588,7 +600,7 @@ public abstract class SqlSecurityTestCase extends ESRestTestCase {
                     if (false == logs.isEmpty()) {
                         fail("Not all logs matched. Unmatched logs:" + logsMessage(logs));
                     }
-                });
+                }, 30, TimeUnit.SECONDS);
             } catch (AssertionError e) {
                 auditFailure = true;
                 logger.warn(

